apps/ledger-live-desktop/src/newArch/hooks/useCopyToClipboard.ts

```diff
@@ -0,0 +1,19 @@
+import { useCallback, useRef } from "react";
+
+export function useCopyToClipboard(callback?: (text: string) => void) {
+  const textRef = useRef<string>();
+
+  const copy = useCallback(() => {
+    const text = textRef.current ?? "";
+    navigator.clipboard.writeText(text).then(() => {
+      if (callback) {
+        callback(text);
+      }
+    });
+  }, [callback]);
+
+  return (text: string) => {
+    textRef.current = text;
+    copy();
```

Comment:
That's kind of a details but here the `useCallback` on copy is still not used because the reference is not returned. It only exist in the returned closure where it's called. This is why I was suggesting to have copy be returned:
```suggestion
    return copy;
```
So it can be passed directly to the component:
```tsx
const handleCopyError = useCopyToClipboard(_text => ...);
...
<InteractFlex onClick={handleCopyError(error.message)}>{icon}</InteractFlex>
```
But again it's a detail and I don't think it should be a blocker. I'll retest and approve the PR if all goes well.